### ISSUE TITLE
Incorrect decoding of branch of size 1 with a value that's already encoded

### ISSUE THREAD

**Issue by paramonov** [fb55/entities#852]
Hello there, I faced a terrible issue using `cheerio` library  
I've created an [issue](https://github.com/inikulin/parse5/issues/581) on parse5 library but after further investigations I realised the problem is here  

I see two different behaviours on legacy entities in the same case on different entities:
```js
const texts = [
    "&cent<", // legacy
    "&nbsp<", // legacy
    "&middot<", // legacy
    "&ensp<", // not legacy
];

for (const text of texts) {
    console.log(`"${text}"`, `"${decodeHTML(text)}"`);
}
```

I'd expect the following output of this:
```console
"&cent<" "¢<"
"&nbsp<" " <"
"&middot<" "·<"
"&ensp<" "&ensp<"
```

but the actual output is:
```console
"&cent<" "¢<"
"&nbsp<" "Á"
"&middot<" "·<"
"&ensp<" "&ensp<"
```

I tried to solve it by my own, but unfortunately need too much time for it due to the lack of byte operations knowledge

**Comment by fb55** [comment]
Thanks for the report! I started looking into this, but haven't gotten far yet. An issue that does need to be fixed desperately though.

**Comment by fb55** [comment]
@paramonov Thanks again for the report! This has been fixed & published as `entities@4.3.1`.
